sql/plan/delete.go

```diff
@@ -91,35 +129,182 @@ func deleteDatabaseHelper(node sql.Node) string {
 	return ""
 }
 
+// Resolved implements the sql.Resolvable interface.
+func (p *DeleteFrom) Resolved() bool {
+	if p.Child.Resolved() == false {
+		return false
+	}
+
+	for _, target := range p.explicitTargets {
+		if target.Resolved() == false {
+			return false
+		}
+	}
+
+	return true
+}
+
 func (p *DeleteFrom) Database() string {
 	return deleteDatabaseHelper(p.Child)
 }
 
+// Validate checks for invalid settings, such as deleting from multiple databases, specifying a delete target table
+// multiple times, or using a DELETE FROM JOIN without specifying any explicit delete target tables, and returns an
+// error if any validation issues were detected.
+func (p *DeleteFrom) Validate() error {
```

Comment:
This should really be in the analyzer's validation phase, but I was hesitant to create a new rule just for validating `DeleteFrom`. I can take another look this afternoon and see if there's a good spot to slide this into an existing analyzer validation rule.

Reply:
i think zach would probably veto this imo. New validation rule is annoying to add, and will need to be excluded from prepared statement preprocessing, but it isn't a high lift and will be skipped for any node that are not deletes. The right way to do validation would have one rule with a visitor call for each individual node type, which will only traverse the tree once. The more complicated version is probably too much for this PR.

Reply:
Yup, definitely feels heavyweight to create many new validation rules for different plan nodes. We could keep grouping them together, but still feels odd. If you aren't concerned with this `Validate` method living here for now, I'm okay with it. 

A more generic first-class concept of node's being able to validate themselves seems like it would be very nice and also seems handy to keep the node-specific validation code closer to the node implementation.